imperial_coldfront_plugin/views.py

```diff
@@ -493,3 +506,95 @@ def group_membership_extend(
         "imperial_coldfront_plugin/extend_membership.html",
         dict(form=form, group_membership=group_membership),
     )
+
+
+PROJECT_ID_PREFIX = "rdf-"
+PROJECT_ID_REGEX = re.compile(PROJECT_ID_PREFIX + r"(?P<number>\d{6})")
+
+
+def get_next_rdf_project_id():
+    """Get the next available RDF project id value."""
+    rdf_id_attribute_type = AllocationAttributeType.objects.get(name="RDF Project ID")
+    group_ids = AllocationAttribute.objects.filter(
+        allocation_attribute_type=rdf_id_attribute_type,
+    ).values_list("value", flat=True)
+    id_numbers = [
+        int(PROJECT_ID_REGEX.match(group_id).groupdict()["number"])
+        for group_id in group_ids
+    ]
+    new_project_number = max(id_numbers) + 1 if id_numbers else 1
+
+    return format_project_number_to_id(new_project_number)
+
+
+def format_project_number_to_id(num):
+    """Format an rdf project number into a full id string."""
+    return f"{PROJECT_ID_PREFIX}{num:06d}"
+
+
+@login_required
+def add_rdf_storage_allocation(request):
+    """Create a new RDF project allocation."""
+    if not request.user.is_superuser:
+        return HttpResponseForbidden()
+
+    if request.method == "POST":
+        form = RDFAllocationForm(request.POST)
+        if form.is_valid():
+            rdf_id_attribute_type = AllocationAttributeType.objects.get(
+                name="RDF Project ID"
+            )
+            group_id = get_next_rdf_project_id()
+            if AllocationAttribute.objects.filter(
+                allocation_attribute_type=rdf_id_attribute_type,
+                value=group_id,
+            ):
+                raise ValueError("RDF project with ID already exists.")
```

Comment:
Seeing the way `get_next_rdf_project_id` works, it is impossible for it to ever return an id that already exist, so this condition will never happen, as far as I can tell. I think this bit can be removed.